src/Hydrator/MoneyHydrator.php

```diff
@@ -20,7 +20,7 @@ class MoneyHydrator implements HydratorInterface
     public function extract($object)
     {
         return [
-            'amount' => $object->getAmount(),
+            'amount' => $object->getAmount() / 100,
```

Comment:
If you want 100 USD, that's 10000 units, you should set 10000 in the form. It wil save 10000 in the database and when you return it, you'll have 10000 in money object.

When you return the money object. The MoneyFormat helper already knows about that and does exactly what you suggested, which is dividing it by 100. But that is a temporary solution, and its expected from the helper to work with non decimal currencies. http://en.wikipedia.org/wiki/Non-decimal_currency